File: solutions/cpp/secret-handshake/secret_handshake.cpp
#include "secret_handshake.h"
#include <algorithm>

namespace secret_handshake {

    auto commands(int code) -> std::vector<std::string>
    {
        std::vector<std::string> actions;

        if (code & 1) actions.push_back("wink");
        if (code & 2) actions.push_back("double blink");
        if (code & 4) actions.push_back("close your eyes");
        if (code & 8) actions.push_back("jump");

        if (code & 16) std::reverse(actions.begin(), actions.end());

        return actions;
    }

}  // namespace secret_handshake

File: solutions/python/secret-handshake/secret_handshake.py
def commands(binary_str):

    actions = []

    if binary_str[4] == "1":
        actions += ["wink"]
    if binary_str[3] == "1":
        actions += ["double blink"]
    if binary_str[2] == "1":
        actions += ["close your eyes"]
    if binary_str[1] == "1":
        actions += ["jump"]

    if binary_str[0] == "1":
        actions.reverse()

    return actions
